ext/datadog_profiling_native_extension/stack_recorder.c

```diff
@@ -481,15 +480,17 @@ static VALUE _native_serialize(DDTRACE_UNUSED VALUE _self, VALUE recorder_instan
   serializer_set_start_timestamp_for_next_profile(state, finish_timestamp);
 
   // Prepare the iteration on heap recorder we'll be doing outside the GVL. The preparation needs to
-  // happen while holding on to the GVL.
-  heap_recorder_prepare_iteration(state->heap_recorder);
+  // happen while holding on to the GVL. Do not include in the iteration objects with age == 0. These
+  // objects that have yet to undergo a GC may just be noise/trash at this point and are usually not
+  // relevant for heap profiles as the great majority should be trivially reclaimed during the next GC.
+  size_t min_age = 1;
+  heap_recorder_prepare_iteration(state->heap_recorder, min_age);
```

Comment:
Minor: Since this constant ends up being repeated in `heap_recorder_prepare_iteration` below, may be worth adding a `#define`?